src/crimson/os/seastore/onode_manager/staged-fltree/node_delta_recorder.h

```diff
@@ -18,7 +18,7 @@ namespace crimson::os::seastore::onode {
 class DeltaRecorder {
  public:
   virtual ~DeltaRecorder() {
-    assert(is_empty());
+    // non-empty DeltaRecorder can be destructed under eagain
```

Comment:
It's actually always ok to dispose of a partial transaction.  eagain simply happens to be a really common case.  Perhaps:

```
// May be non-empty if transaction is abandoned without being submitted -- conflicts are a particularly common
// example (denoted generally by returning crimson::ct_error::eagain).